site/content/integrate/apps/functionality/kv-store.md

```diff
@@ -4,19 +4,99 @@ description: Use the per-App key-value data store
 
-There are 3 operations that can be performed against the key-value store:
+There are three operations that can be performed against the key-value store:
 
-1. Get the value of a key (`KVGet`)
-2. Set the value of a key (`KVSet`)
-3. Delete a key (`KVDelete`)
+1. Get the value of a key
+2. Set the value of a key
+3. Delete a key
 
-For example, using the [Golang driver]({{<ref "/integrate/apps/drivers/golang">}}) to store a string value would look like the following:
+The key-value store can be accessed in two ways:
+
+- Using HTTP REST endpoints
+- Using a [driver]({{<ref "/integrate/apps/drivers">}})
+
+{{<note "Unique keys:">}}
+Keys in the Key-value store are unique to the user or bot token used. One user's data will not overwrite another user's data. 
+{{</note>}}
+
+## Use HTTP REST endpoints
+
+### Endpoint URL
+
+The key-value store is accessed using a single HTTP REST endpoint:
+
+`<mattermost_site_url>/plugins/com.mattermost.apps/api/v1/kv/<my-key>`
+
+Replace `<mattermost_site_url>` with the base URL to the Mattermost server and `<my-key>` with the name of the key to add, modify, or delete.
+The `<mattermost_site_url>` value can be obtained from a [call request context]({{<ref "/integrate/apps/structure/call#context">}}).
+
+#### Authorization
+
+An authorization token is required when invoking HTTP REST endpoints of the key-value store. The token must be set in the `Authorization` header as a bearer token.
+The acting user token can be obtained from a call request context where the `acting_user_access_token` expand field is set to `all`.
+The `acting_user_access_token` field of the call request context will contain the token.
+
+### Get a value with HTTP REST
+
+Example HTTP request to get the value of key `my-key`:
+
+```http request
+GET /plugins/com.mattermost.apps/api/v1/kv/my-key HTTP/1.1
+Authorization: Bearer xxxxxxxxxxxxxx
+```
+
+The HTTP response body will contain only the value of the requested key.
+
+### Set a value with HTTP REST
+
+Example HTTP request to set the value of key `my-key` to a JSON object:
+
+```http request
+POST /plugins/com.mattermost.apps/api/v1/kv/my-key HTTP/1.1
+Authorization: Bearer xxxxxxxxxxxxxx
+Content-Type: application/json
+
+{
+    "object_with": "some_data"
+}
+```
+
+### Delete a value with HTTP REST
+
+Example HTTP request to delete the key `my-key`:
+
+```http request
+DELETE /plugins/com.mattermost.apps/api/v1/kv/my-key HTTP/1.1
+Authorization: Bearer xxxxxxxxxxxxxx
+```
+
+## Use a driver
+
+### Get a value with a driver
+
+Example [Golang driver]({{<ref "/integrate/apps/drivers/golang">}}) code to get the value of the key `my-key`:
+
+```go
+// Create an instance of the REST API client, acting as a bot (vs. a user)
+client := appclient.AsBot(callRequest.Context)
+// Create an empty struct to hold the data
+var myData string
+// Retrieve the value of the key `my-key` into `myData`
+err := client.KVGet("my-app", "my-key", &myData)
+if err != nil {
+    // handle the error
+}
```

Comment:
It should be noted that `myData` can be any type that can be unmarshaled from JSON. The `KVGet` method automatically does the unmarshaling for you.